Userland/Libraries/LibAudio/Buffer.h

```diff
@@ -67,19 +71,20 @@ class ResampleHelper {
 // A buffer of audio samples.
 class Buffer : public RefCounted<Buffer> {
 public:
-    static NonnullRefPtr<Buffer> from_pcm_data(ReadonlyBytes data, int num_channels, PcmSampleFormat sample_format);
-    static NonnullRefPtr<Buffer> from_pcm_stream(InputMemoryStream& stream, int num_channels, PcmSampleFormat sample_format, int num_samples);
-    static NonnullRefPtr<Buffer> create_with_samples(Vector<Sample>&& samples)
+    static ErrorOr<NonnullRefPtr<Buffer>> from_pcm_data(ReadonlyBytes data, int num_channels, PcmSampleFormat sample_format);
+    static ErrorOr<NonnullRefPtr<Buffer>> from_pcm_stream(InputMemoryStream& stream, int num_channels, PcmSampleFormat sample_format, int num_samples);
+    static ErrorOr<NonnullRefPtr<Buffer>> create_with_samples(Vector<Sample>&& samples)
     {
-        return adopt_ref(*new Buffer(move(samples)));
+        return adopt_nonnull_ref_or_enomem(new (nothrow) Buffer(move(samples)));
     }
-    static NonnullRefPtr<Buffer> create_with_anonymous_buffer(Core::AnonymousBuffer buffer, i32 buffer_id, int sample_count)
+    static ErrorOr<NonnullRefPtr<Buffer>> create_with_anonymous_buffer(Core::AnonymousBuffer buffer, i32 buffer_id, int sample_count)
     {
-        return adopt_ref(*new Buffer(move(buffer), buffer_id, sample_count));
+        return adopt_nonnull_ref_or_enomem(new (nothrow) Buffer(move(buffer), buffer_id, sample_count));
     }
     static NonnullRefPtr<Buffer> create_empty()
     {
-        return adopt_ref(*new Buffer({}));
+        // If we can't allocate an empty buffer, things are in a very bad state.
+        return MUST(adopt_nonnull_ref_or_enomem(new (nothrow) Buffer({})));
```

Comment:
this bit should go in the previous commit, though I guess it's next to related changes...

Reply:
I'd keep it here as it fits with the error API changes.